### ISSUE TITLE
Failed downgrading Papirus icon theme due to epapirus package conflict

### ISSUE THREAD

**Issue by neopostmodern** [pop-os/upgrade#337]
Here's the relevant commands & output
```
❯ pop-upgrade release upgrade -f
checking if pop-upgrade requires an update
Current Release: 22.04
Upgrading to: 24.04
New version available: 10
Event: updating package lists
Event: upgrading packages for the current release
Event: fetching updated packages for the current release
Release upgrade status: release upgrade aborted: failed to downgrade packages

took 1m25s
❯ lsb_release -a
No LSB modules are available.
Distributor ID:	Pop
Description:	Pop!_OS 22.04 LTS
Release:	22.04
Codename:	jammy
```

**Comment by jacobgkau** [comment]
After seeing that error, please run `sudo journalctl -u pop-upgrade -b 0 > upgrade-log.txt` (this needs to be done before the next time you reboot), then attach the file to a comment here.

**Comment by jacobgkau** [comment]
If you've already rebooted since you got the error, then just run `sudo journalctl -u pop-upgrade > upgrade-log.txt` to create the log. It will be unnecessarily large, but will include all of the available history.

**Comment by neopostmodern** [comment]
Here's the log 
<details>
<summary>upgrade-log.txt</summary>
```
Okt 16 11:10:47 cyber systemd[1]: Starting Pop Upgrade Daemon...
Okt 16 11:10:47 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:403: initializing daemon
Okt 16 11:10:47 cyber systemd[1]: Started Pop Upgrade Daemon.
Okt 16 11:10:47 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:792: daemon registered -- listening for new events
Okt 16 11:10:47 cyber pop-upgrade[17341]: pop-upgrade was already not on hold.
Okt 16 11:11:05 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:1144: updating apt sources
Okt 16 11:11:05 cyber pop-upgrade[21922]: Hit:1 http://apt.pop-os.org/proprietary jammy InRelease
Okt 16 11:11:05 cyber pop-upgrade[21922]: Hit:2 http://apt.pop-os.org/release jammy InRelease
Okt 16 11:11:05 cyber pop-upgrade[21922]: Hit:3 http://apt.pop-os.org/ubuntu jammy InRelease
Okt 16 11:11:06 cyber pop-upgrade[21922]: Hit:4 http://apt.pop-os.org/ubuntu jammy-security InRelease
Okt 16 11:11:06 cyber pop-upgrade[21922]: Hit:5 http://apt.pop-os.org/ubuntu jammy-updates InRelease
Okt 16 11:11:06 cyber pop-upgrade[21922]: Hit:6 http://apt.pop-os.org/ubuntu jammy-backports InRelease
Okt 16 11:11:08 cyber pop-upgrade[21922]: Reading package lists...
Okt 16 11:11:08 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:1055: performing a release check
Okt 16 11:11:08 cyber pop-upgrade[17331]: [INFO ] release_api.rs:58: checking for build 24.04 in channel intel
Okt 16 11:11:09 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:1062: Release { current: "22.04", lts: "true",  next: "24.04", available: 10 }
Okt 16 11:11:09 cyber pop-upgrade[17331]: [INFO ] release_api.rs:58: checking for build 22.04 in channel nvidia
Okt 16 11:11:10 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:1130: setting status to upgrading distribution release
Okt 16 11:11:10 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:1080: upgrading release from 22.04 to 24.04, with 1
Okt 16 11:11:10 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:339: attempting release upgrade, using a offline upgrade
Okt 16 11:11:11 cyber pop-upgrade[23841]: Canceled hold on pop-system-updater.
Okt 16 11:11:11 cyber pop-upgrade[23971]: pop-upgrade set on hold.
Okt 16 11:11:11 cyber pop-upgrade[23971]: pop-system-updater set on hold.
Okt 16 11:11:12 cyber pop-upgrade[24156]: Hit:1 http://apt.pop-os.org/proprietary jammy InRelease
Okt 16 11:11:13 cyber pop-upgrade[24156]: Hit:2 http://apt.pop-os.org/release jammy InRelease
Okt 16 11:11:13 cyber pop-upgrade[24156]: Hit:3 http://apt.pop-os.org/ubuntu jammy InRelease
Okt 16 11:11:14 cyber pop-upgrade[24156]: Hit:4 http://apt.pop-os.org/ubuntu jammy-security InRelease
Okt 16 11:11:14 cyber pop-upgrade[24156]: Hit:5 http://apt.pop-os.org/ubuntu jammy-updates InRelease
Okt 16 11:11:14 cyber pop-upgrade[24156]: Hit:6 http://apt.pop-os.org/ubuntu jammy-backports InRelease
Okt 16 11:11:16 cyber pop-upgrade[24156]: Reading package lists...
Okt 16 11:11:16 cyber pop-upgrade[25778]: Reading package lists...
Okt 16 11:11:16 cyber pop-upgrade[25778]: Building dependency tree...
Okt 16 11:11:16 cyber pop-upgrade[25778]: Reading state information...
Okt 16 11:11:16 cyber pop-upgrade[25778]: 0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
Okt 16 11:11:17 cyber pop-upgrade[17331]: [INFO ] repair/packaging.rs:80: ensuring prerequisites are installed
Okt 16 11:11:17 cyber pop-upgrade[17331]: [INFO ] repair/packaging.rs:98: installing required prerequisites: []
Okt 16 11:11:17 cyber pop-upgrade[25964]: Reading package lists...
Okt 16 11:11:17 cyber pop-upgrade[25964]: Building dependency tree...
Okt 16 11:11:17 cyber pop-upgrade[25964]: Reading state information...
Okt 16 11:11:18 cyber pop-upgrade[25964]: 0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:436: creating backup of source lists
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/system.sources.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/docker.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/teamviewer.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/google-chrome.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/papirus-ubuntu-papirus-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/alex-p-ubuntu-aegisub-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/mongodb-org-6.0.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/google-earth-pro.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/pop-os-apps.sources.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/helkaluin-ubuntu-webp-pixbuf-loader-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/microsoft-edge.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/pop-os-release.sources.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:66: removing old backup at /etc/apt/sources.list.d/alex-p-ubuntu-tesseract-ocr5-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list to /etc/apt/sources.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/alex-p-ubuntu-aegisub-jammy.list to /etc/apt/sources.list.d/alex-p-ubuntu-aegisub-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/helkaluin-ubuntu-webp-pixbuf-loader-jammy.list to /etc/apt/sources.list.d/helkaluin-ubuntu-webp-pixbuf-loader-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/pop-os-release.sources to /etc/apt/sources.list.d/pop-os-release.sources.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/google-chrome.list to /etc/apt/sources.list.d/google-chrome.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/alex-p-ubuntu-tesseract-ocr5-jammy.list to /etc/apt/sources.list.d/alex-p-ubuntu-tesseract-ocr5-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/system.sources to /etc/apt/sources.list.d/system.sources.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/mongodb-org-6.0.list to /etc/apt/sources.list.d/mongodb-org-6.0.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/google-earth-pro.list to /etc/apt/sources.list.d/google-earth-pro.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/papirus-ubuntu-papirus-jammy.list to /etc/apt/sources.list.d/papirus-ubuntu-papirus-jammy.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/pop-os-apps.sources to /etc/apt/sources.list.d/pop-os-apps.sources.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/microsoft-edge.list to /etc/apt/sources.list.d/microsoft-edge.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/docker.list to /etc/apt/sources.list.d/docker.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:77: creating backup of /etc/apt/sources.list.d/teamviewer.list to /etc/apt/sources.list.d/teamviewer.list.save
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:544: disabling third party sources
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/alex-p-ubuntu-aegisub-jammy.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/helkaluin-ubuntu-webp-pixbuf-loader-jammy.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/google-chrome.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/alex-p-ubuntu-tesseract-ocr5-jammy.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/mongodb-org-6.0.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/google-earth-pro.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/papirus-ubuntu-papirus-jammy.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/microsoft-edge.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/docker.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:113: disabling sources in /etc/apt/sources.list.d/teamviewer.list
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] release/repos.rs:288: creating source repository files for impish+
Okt 16 11:11:18 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: ReleaseUpgradeEvent(
Okt 16 11:11:18 cyber pop-upgrade[17331]:     UpdatingPackageLists,
Okt 16 11:11:18 cyber pop-upgrade[17331]: )
Okt 16 11:11:18 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: release upgrade: updating package lists
Okt 16 11:11:18 cyber pop-upgrade[26320]: Hit:1 http://apt.pop-os.org/proprietary jammy InRelease
Okt 16 11:11:18 cyber pop-upgrade[26320]: Hit:2 http://apt.pop-os.org/release jammy InRelease
Okt 16 11:11:19 cyber pop-upgrade[26320]: Hit:3 http://apt.pop-os.org/ubuntu jammy InRelease
Okt 16 11:11:19 cyber pop-upgrade[26320]: Hit:4 http://apt.pop-os.org/ubuntu jammy-security InRelease
Okt 16 11:11:19 cyber pop-upgrade[26320]: Hit:5 http://apt.pop-os.org/ubuntu jammy-updates InRelease
Okt 16 11:11:19 cyber pop-upgrade[26320]: Hit:6 http://apt.pop-os.org/ubuntu jammy-backports InRelease
Okt 16 11:11:21 cyber pop-upgrade[26320]: Reading package lists...
Okt 16 11:11:21 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: ReleaseUpgradeEvent(
Okt 16 11:11:21 cyber pop-upgrade[17331]:     UpgradingPackages,
Okt 16 11:11:21 cyber pop-upgrade[17331]: )
Okt 16 11:11:21 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: release upgrade: upgrading packages for the current release
Okt 16 11:11:23 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Fetching(
Okt 16 11:11:23 cyber pop-upgrade[17331]:     "http://apt.pop-os.org/ubuntu/pool/main/liba/libarchive/libarchive13_3.6.0-1ubuntu1.2_amd64.deb",
Okt 16 11:11:23 cyber pop-upgrade[17331]: )
Okt 16 11:11:23 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Fetched(
Okt 16 11:11:23 cyber pop-upgrade[17331]:     "libarchive13_3.6.0-1ubuntu1.2_amd64.deb",
Okt 16 11:11:23 cyber pop-upgrade[17331]:     1,
Okt 16 11:11:23 cyber pop-upgrade[17331]:     1,
Okt 16 11:11:23 cyber pop-upgrade[17331]: )
Okt 16 11:11:23 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: ReleaseUpgradeEvent(
Okt 16 11:11:23 cyber pop-upgrade[17331]:     FetchingPackages,
Okt 16 11:11:23 cyber pop-upgrade[17331]: )
Okt 16 11:11:23 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: release upgrade: fetching updated packages for the current release
Okt 16 11:11:23 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:372: autoremoving packages
Okt 16 11:11:23 cyber pop-upgrade[28127]: Reading package lists...
Okt 16 11:11:23 cyber pop-upgrade[28127]: Building dependency tree...
Okt 16 11:11:23 cyber pop-upgrade[28127]: Reading state information...
Okt 16 11:11:24 cyber pop-upgrade[28127]: 0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
Okt 16 11:11:24 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:361: upgrading packages
Okt 16 11:11:27 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:27 cyber pop-upgrade[17331]:     Progress {
Okt 16 11:11:27 cyber pop-upgrade[17331]:         percent: 0,
Okt 16 11:11:27 cyber pop-upgrade[17331]:     },
Okt 16 11:11:27 cyber pop-upgrade[17331]: )
Okt 16 11:11:27 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: progress: [  0%]
Okt 16 11:11:27 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:27 cyber pop-upgrade[17331]:     Progress {
Okt 16 11:11:27 cyber pop-upgrade[17331]:         percent: 20,
Okt 16 11:11:27 cyber pop-upgrade[17331]:     },
Okt 16 11:11:27 cyber pop-upgrade[17331]: )
Okt 16 11:11:27 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: progress: [ 20%]
Okt 16 11:11:27 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:27 cyber pop-upgrade[17331]:     Unpacking {
Okt 16 11:11:27 cyber pop-upgrade[17331]:         package: "libarchive13:amd64",
Okt 16 11:11:27 cyber pop-upgrade[17331]:         version: "3.6.0-1ubuntu1.2",
Okt 16 11:11:27 cyber pop-upgrade[17331]:         over: "3.6.0-1ubuntu1.1",
Okt 16 11:11:27 cyber pop-upgrade[17331]:     },
Okt 16 11:11:27 cyber pop-upgrade[17331]: )
Okt 16 11:11:27 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: unpacking libarchive13:amd64 (3.6.0-1ubuntu1.2) over (3.6.0-1ubuntu1.1)
Okt 16 11:11:27 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:27 cyber pop-upgrade[17331]:     Progress {
Okt 16 11:11:27 cyber pop-upgrade[17331]:         percent: 40,
Okt 16 11:11:27 cyber pop-upgrade[17331]:     },
Okt 16 11:11:27 cyber pop-upgrade[17331]: )
Okt 16 11:11:27 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: progress: [ 40%]
Okt 16 11:11:28 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:28 cyber pop-upgrade[17331]:     SettingUp {
Okt 16 11:11:28 cyber pop-upgrade[17331]:         package: "libarchive13:amd64",
Okt 16 11:11:28 cyber pop-upgrade[17331]:     },
Okt 16 11:11:28 cyber pop-upgrade[17331]: )
Okt 16 11:11:28 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: setting up libarchive13:amd64
Okt 16 11:11:28 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:28 cyber pop-upgrade[17331]:     Progress {
Okt 16 11:11:28 cyber pop-upgrade[17331]:         percent: 60,
Okt 16 11:11:28 cyber pop-upgrade[17331]:     },
Okt 16 11:11:28 cyber pop-upgrade[17331]: )
Okt 16 11:11:28 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: progress: [ 60%]
Okt 16 11:11:28 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:28 cyber pop-upgrade[17331]:     Progress {
Okt 16 11:11:28 cyber pop-upgrade[17331]:         percent: 80,
Okt 16 11:11:28 cyber pop-upgrade[17331]:     },
Okt 16 11:11:28 cyber pop-upgrade[17331]: )
Okt 16 11:11:28 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: progress: [ 80%]
Okt 16 11:11:28 cyber pop-upgrade[17331]: [DEBUG] daemon/mod.rs:867: Sending DBus Event: Upgrade(
Okt 16 11:11:28 cyber pop-upgrade[17331]:     Processing {
Okt 16 11:11:28 cyber pop-upgrade[17331]:         package: "libc-bin",
Okt 16 11:11:28 cyber pop-upgrade[17331]:     },
Okt 16 11:11:28 cyber pop-upgrade[17331]: )
Okt 16 11:11:28 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:874: package upgrade: processing triggers for libc-bin
Okt 16 11:11:28 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:401: autoremoving packages
Okt 16 11:11:28 cyber pop-upgrade[29183]: Reading package lists...
Okt 16 11:11:28 cyber pop-upgrade[29183]: Building dependency tree...
Okt 16 11:11:28 cyber pop-upgrade[29183]: Reading state information...
Okt 16 11:11:29 cyber pop-upgrade[29183]: 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Okt 16 11:11:29 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:500: searching for packages that require downgrading
Okt 16 11:11:35 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:518: downgrading packages with: DEBIAN_FRONTEND="noninteractive" LANG="C" "apt-get" "--allow-downgrades" "-y" "install" "papirus-icon-theme=20220204-1"
Okt 16 11:11:35 cyber pop-upgrade[30915]: Reading package lists...
Okt 16 11:11:35 cyber pop-upgrade[30915]: Building dependency tree...
Okt 16 11:11:35 cyber pop-upgrade[30915]: Reading state information...
Okt 16 11:11:35 cyber pop-upgrade[30915]: The following packages were automatically installed and are no longer required:
Okt 16 11:11:35 cyber pop-upgrade[30915]:   epapirus-icon-theme papirus-folders
Okt 16 11:11:35 cyber pop-upgrade[30915]: Use 'apt autoremove' to remove them.
Okt 16 11:11:35 cyber pop-upgrade[30915]: The following packages will be DOWNGRADED:
Okt 16 11:11:35 cyber pop-upgrade[30915]:   papirus-icon-theme
Okt 16 11:11:36 cyber pop-upgrade[30915]: 0 upgraded, 0 newly installed, 1 downgraded, 0 to remove and 0 not upgraded.
Okt 16 11:11:36 cyber pop-upgrade[30915]: Need to get 0 B/21,7 MB of archives.
Okt 16 11:11:36 cyber pop-upgrade[30915]: After this operation, 11,1 MB of additional disk space will be used.
Okt 16 11:11:36 cyber pop-upgrade[30915]: dpkg: warning: downgrading papirus-icon-theme from 20240201-6491+pkg29~ubuntu22.04.1 to 20220204-1
Okt 16 11:11:37 cyber pop-upgrade[30915]: [615B blob data]
Okt 16 11:11:37 cyber pop-upgrade[30915]: Preparing to unpack .../papirus-icon-theme_20220204-1_all.deb ...
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]: WARNING!
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]: We strongly do not recommend to downgrade this package in this way.
Okt 16 11:11:37 cyber pop-upgrade[30915]: Please remove this package and then install it again:
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]:   sudo apt-get remove papirus-icon-theme
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]: Unpacking papirus-icon-theme (20220204-1) over (20240201-6491+pkg29~ubuntu22.04.1) ...
Okt 16 11:11:45 cyber pop-upgrade[30915]: dpkg: error processing archive /var/cache/apt/archives/papirus-icon-theme_20220204-1_all.deb (--unpack):
Okt 16 11:11:45 cyber pop-upgrade[30915]:  trying to overwrite '/usr/share/icons/ePapirus/16x16/actions/action-unavailable.svg', which is also in package epapirus-icon-theme 20240201-6491+pkg29~ubuntu22.04.1
Okt 16 11:11:47 cyber pop-upgrade[30915]: dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Okt 16 11:12:01 cyber pop-upgrade[30915]: Errors were encountered while processing:
Okt 16 11:12:01 cyber pop-upgrade[30915]:  /var/cache/apt/archives/papirus-icon-theme_20220204-1_all.deb
Okt 16 11:12:01 cyber pop-upgrade[30915]: E: Sub-process /usr/bin/dpkg returned an error code (1)
Okt 16 11:12:01 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:362: upgrade result: Err(Downgrade(apt-get downgrade
Okt 16 11:12:01 cyber pop-upgrade[17331]: Caused by:
Okt 16 11:12:01 cyber pop-upgrade[17331]:     status is unknown: exit status: 100))
Okt 16 11:12:01 cyber pop-upgrade[37166]: Canceled hold on pop-upgrade.
Okt 16 11:12:01 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:366: setting upgrade state
```
</details>

By manually removing the offending package I could solve the issue and now the upgrader seems to run fine. So I guess it's not really a fault with the upgrader, but rather a feedback / legible output issue?

**Comment by jacobgkau** [comment]
Thank you for the output. So it's something to do with the Papirus icon theme:

<details>
<summary>Relevant output:</summary>

```
Okt 16 11:11:29 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:500: searching for packages that require downgrading
Okt 16 11:11:35 cyber pop-upgrade[17331]: [INFO ] release/mod.rs:518: downgrading packages with: DEBIAN_FRONTEND="noninteractive" LANG="C" "apt-get" "--allow-downgrades" "-y" "install" "papirus-icon-theme=20220204-1"
Okt 16 11:11:35 cyber pop-upgrade[30915]: Reading package lists...
Okt 16 11:11:35 cyber pop-upgrade[30915]: Building dependency tree...
Okt 16 11:11:35 cyber pop-upgrade[30915]: Reading state information...
Okt 16 11:11:35 cyber pop-upgrade[30915]: The following packages were automatically installed and are no longer required:
Okt 16 11:11:35 cyber pop-upgrade[30915]:   epapirus-icon-theme papirus-folders
Okt 16 11:11:35 cyber pop-upgrade[30915]: Use 'apt autoremove' to remove them.
Okt 16 11:11:35 cyber pop-upgrade[30915]: The following packages will be DOWNGRADED:
Okt 16 11:11:35 cyber pop-upgrade[30915]:   papirus-icon-theme
Okt 16 11:11:36 cyber pop-upgrade[30915]: 0 upgraded, 0 newly installed, 1 downgraded, 0 to remove and 0 not upgraded.
Okt 16 11:11:36 cyber pop-upgrade[30915]: Need to get 0 B/21,7 MB of archives.
Okt 16 11:11:36 cyber pop-upgrade[30915]: After this operation, 11,1 MB of additional disk space will be used.
Okt 16 11:11:36 cyber pop-upgrade[30915]: dpkg: warning: downgrading papirus-icon-theme from 20240201-6491+pkg29~ubuntu22.04.1 to 20220204-1
Okt 16 11:11:37 cyber pop-upgrade[30915]: [615B blob data]
Okt 16 11:11:37 cyber pop-upgrade[30915]: Preparing to unpack .../papirus-icon-theme_20220204-1_all.deb ...
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]: WARNING!
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]: We strongly do not recommend to downgrade this package in this way.
Okt 16 11:11:37 cyber pop-upgrade[30915]: Please remove this package and then install it again:
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]:   sudo apt-get remove papirus-icon-theme
Okt 16 11:11:37 cyber pop-upgrade[30915]: 
Okt 16 11:11:37 cyber pop-upgrade[30915]: Unpacking papirus-icon-theme (20220204-1) over (20240201-6491+pkg29~ubuntu22.04.1) ...
Okt 16 11:11:45 cyber pop-upgrade[30915]: dpkg: error processing archive /var/cache/apt/archives/papirus-icon-theme_20220204-1_all.deb (--unpack):
Okt 16 11:11:45 cyber pop-upgrade[30915]:  trying to overwrite '/usr/share/icons/ePapirus/16x16/actions/action-unavailable.svg', which is also in package epapirus-icon-theme 20240201-6491+pkg29~ubuntu22.04.1
Okt 16 11:11:47 cyber pop-upgrade[30915]: dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Okt 16 11:12:01 cyber pop-upgrade[30915]: Errors were encountered while processing:
Okt 16 11:12:01 cyber pop-upgrade[30915]:  /var/cache/apt/archives/papirus-icon-theme_20220204-1_all.deb
Okt 16 11:12:01 cyber pop-upgrade[30915]: E: Sub-process /usr/bin/dpkg returned an error code (1)
Okt 16 11:12:01 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:362: upgrade result: Err(Downgrade(apt-get downgrade
Okt 16 11:12:01 cyber pop-upgrade[17331]: Caused by:
Okt 16 11:12:01 cyber pop-upgrade[17331]:     status is unknown: exit status: 100))
Okt 16 11:12:01 cyber pop-upgrade[37166]: Canceled hold on pop-upgrade.
Okt 16 11:12:01 cyber pop-upgrade[17331]: [INFO ] daemon/mod.rs:366: setting upgrade state
```
</details>

We definitely want the upgrader to be able to handle things like this without user intervention, if possible. If need be, we could get as specific as checking if this package is installed and if it is, removing and then installing it instead of trying to directly downgrade. It's very strange that that would be required, though.

I see you had version `20240201-6491+pkg29~ubuntu22.04.1`, which appears to come from [that project's PPA](https://launchpad.net/~papirus/+archive/ubuntu/papirus?field.series_filter=jammy). As expected, the version being downgraded to was 20220204-1, which is in the [Ubuntu repository](https://packages.ubuntu.com/jammy/papirus-icon-theme). But there was a conflict with another package, `epapirus-icon-theme`, which I'm not currently seeing listed anywhere. From the [project's readme](https://github.com/PapirusDevelopmentTeam/papirus-icon-theme?tab=readme-ov-file#ubuntu-and-derivatives), the version with "e" in the front is/was intended to be used for elementary OS only. It [seems](https://github.com/PapirusDevelopmentTeam/papirus-icon-theme/issues/3563) the package for epapirus was split in October of last year-- did you install Papirus before or after that?

<details>
<summary>I found this in the Papirus changelog:</summary>

> **For PPA users**
> 
> In most cases, you will not see the difference. If recommended packages are not denied on your system, `epapirus-icon-theme` package will be installed as before, but now you can remove ePapirus and ePapirus-Dark icon theme using `sudo apt remove epapirus-icon-theme`, in case you're not an elementaryOS user.
> 
> **For package maintainers**
> 
> The default behavior didn't change. The Makefile still installs all the available icon themes. But we recommend two options to deal with ePapirus and ePapirus-Dark themes:
> 
> 1. If your distro supports [Pantheon Desktop](https://en.wikipedia.org/wiki/Elementary_OS#Pantheon_desktop_environment) please split the package to `papirus-icon-theme` (Papirus, Papirus-Dark, Papirus-Light), and `epapirus-icon-theme` (ePapirus, ePapirus-Dark).  epapirus-icon-theme should have `papirus-icon-theme` as dependency. See example for [deb package](https://github.com/PapirusDevelopmentTeam/papirus-icon-theme/commit/635c7bc416edd41b028a328df61ffae4cf053a7c).
> 2. If your distro does **NOT** support [Pantheon Desktop](https://en.wikipedia.org/wiki/Elementary_OS#Pantheon_desktop_environment), please exclude ePapirus and ePapirus-Dark from `papirus-icon-theme` package using EXCLUDE notation: `make EXCLUDE="ePapirus ePapirus-Dark" install`.

</details>

Based on that, it sounds like maybe we can just automatically try to remove the epapirus packages if it's installed, leaving only the normal papirus package. The Ubuntu package seemingly still includes the epapirus icons (it doesn't look like they've split the package even in 24.04), so the icons will still end up being installed (even though they're not meant to be used outside of elementary OS), but this way, it won't have a file conflict with the PPA's epapirus package, which the Ubuntu package should be marked as conflicting with but isn't.

**Comment by neopostmodern** [comment]
Likely before then, I think I've used that icon theme for quite some time. My notes would estimate it to July 2022.
(Also I got tired of PPAs so I regularly install single DEBs, might have been the case here too, but according to my notes I used the PPA properly.)

**Comment by jacobgkau** [comment]
I was able to replicate this issue in a virtual machine. All I needed to do was install `papirus-icon-theme` from the PPA before attempting to upgrade. Even though `epapirus-icon-theme` is only intended for elementary OS, `papirus-icon-theme` pulls it in as a recommends by default, so manual user intervention would be required for it to not be present.

(The Papirus readme is somewhat misleading with its suggestion for elementary OS users to install `epapirus-icon-theme` separately; the way their packages are currently set up allow elementary OS users to avoid installing the unnecessary main package, but result in users of all other distros getting the unnecessary elementary package installed by default.)

![screenshot-2024-10-17-20-41-26](https://github.com/user-attachments/assets/dac9fcd1-648e-4145-9718-33be03265c61)

In `daemon/src/release/mod.rs`, before or within the `downgrade_packages()` function, we need to check if `epapirus-icon-theme` is installed and, if so, remove it.
